apps/backend/src/deposits/pos-deposit.service.ts

```diff
@@ -160,4 +166,34 @@ export class PosDepositService {
   ): Promise<POSDepositEntity[]> {
     return await this.posDepositRepo.save(deposits);
   }
+
+  /**
+   * updateDepositStatus
+   * This will update necessary deposits to a desired MatchStatus.
+   * This should be more performant as it is wrapped in a transaction
+   * @param deposits A list of deposit entities to set a new status for, with the expected heuristic round
+   * @returns {void}
+   */
+  async updateDepositStatus(
+    deposits: POSDepositEntity[],
+    status: MatchStatus
+  ): Promise<void> {
+    return this.posDepositRepo.manager.transaction(async (manager) => {
```

Comment:
Nice, this is great. 

Can we return the UpdateResult type from typeorm instead of void? (Since we're calling return here.) I just think it will be more log-able/test-able this way.

Reply:
...also, I think we don't need to pass the status since we're setting the status earlier during the match process? Either we should pass in all the new values (ie status, heuristic round), or, destructure them all from the entity itself? (unless there is a reason for doing it this way that I am missing?)

Reply:
Will address both